### DISCUSSION TITLE
How do I install lldb-vscode？

### DISCUSSION THREAD

**Discussion by watjt** [helix-editor/helix#5297] [Q&A]
hx --health
rust                             ✓ rust-analyzer                  ✘ lldb-vscode

I just finded this [codelldb-x86_64-linux.vsix](https://github.com/vadimcn/vscode-lldb/releases/download/v1.8.1/codelldb-x86_64-linux.vsix)

**Comment by sudormrfbin** [comment]
`lldb-vscode` is used as a debugger, and debugging support in helix is experimental for now. You can find install instructions in the [wiki][1]. Related: https://github.com/helix-editor/helix/discussions/4048.

[1]: https://github.com/helix-editor/helix/wiki/Debugger-Configurations#install-debuggers

**Reply by watjt** [reply]
I have another problem, I installed helix editor, but no auto-complete when write rust or go code. How can i configurate?

hx --health
rust                                  ✓ rust-analyzer                   ✘ lldb-vscode                   ✓                                     ✓                                     ✓
go                                    ✓ gopls                               ✓ dlv                                 ✓                                     ✓                                     ✓

**Reply by CalciteW** [reply]
Install rust-analyzer ：
https://github.com/helix-editor/helix/wiki/How-to-install-the-default-language-servers#rust_analyzer

**Reply by watjt** [reply]
It has been installed. But not have auto-complete
✓ rust-analyzer

**Reply by watjt** [reply]
My development environment is intranet, so I use this package https://github.com/helix-editor/helix/releases/download/22.12/helix-22.12-x86_64-linux.tar.xz  to install offline, is it because of offline installation?

**Reply by CalciteW** [reply]
Sorry, I don't know. I installed it with the package manager.

**Reply by CalciteW** [reply]
You can press `<Space-k>` on the variable to see if the document appears, if it doesn't, the LSP is not working properly

**Reply by CalciteW** [reply]
Try this https://github.com/helix-editor/helix/discussions/4737

**Comment by Quorsor** [comment]
I don't know if it's really the reason why it worked but I just used chocolatey to install LLVM and it somehow gave me a check. I was actually trying to install codelldb since I have nothing to lead me on where or how I can do lldb-vscode.

`choco install llvm`

OS: Windows 11
Terminal: Powershell

**Reply by jerhos** [reply]
Just wanted to put this here for visibility.
Chocolatey didn't work for me, but WinGet did: 

`winget install LLVM.LLVM`

This also adds it to the path - of which I'm not quite sure if choco does this. You might want to restart the terminal before it's recognized.

**Comment by SmiteWindows** [comment]
ame i llvm
paru -S llvm
yay -S llvm
pacman -S llvm

**Reply by David-Else** [reply]
I just added this `pacman` instruction to the WIKI, can you confirm this is all you need to do to make it work in Helix? https://github.com/helix-editor/helix/wiki/Debugger-Configurations#for-arch-linux-users

**Reply by hackers267** [reply]
`paru -S llvm` doesn't work for me, But `paru -S lldb` works for me.

**Reply by David-Else** [reply]
I have never used `pacman`, please could you update the WIKI with whatever works, I was just copy pasting from the comment above.

**Reply by Familex** [reply]
`pacman -S lldb` worked for me.

**Reply by gaivenis** [reply]
@hackers267, `paru -S lldb` worked for me

**Reply by hackers267** [reply]
@David-Else Thanks, I had updated the WIKI about *how to install the `vscode-lldb` for archlinux user*

**Comment by HealsCodes** [comment]
for macOS (using homebrew):

```zsh
brew install llvm
ln -s $(brew --prefix)/opt/llvm/bin/lldb-vscode $(brew --prefix)/bin/
```

The second line is required to get lldb-vscode into your homebrew PATH without exposing the whole lldb installation as it would clash with xcode commandline tools.

**Reply by abelcha** [reply]
btw if you have linked a llvm version greater then 16:
`brew install llvm@16`
`ln -s $(brew --prefix)/opt/llvm@16/bin/lldb-vscode $(brew --prefix)/bin/`

**Comment by nuke-web3** [comment]
For Debian/Ubuntu, I needed to install and further link to get things working:

```sh
sudo bash -c "$(wget -O - https://apt.llvm.org/llvm.sh)" # Instructions from https://apt.llvm.org/
cd $(dirname $(which lldb-17))
sudo ln -s lldb-vscode-17 lldb-vscode 
```

**Reply by David-Else** [reply]
@NukeManDan  I am trying to avoid installing nightly llvm on Debian 12. I tried `sudo apt install llvm-15` but it does not seem to contain any permutation of `lldb-vscode`. Did you discover any `lldb-vscode` inside of any official Debian packages? Cheers!

**Comment by HealsCodes** [comment]
> @NukeManDan I am trying to avoid installing nightly llvm on Debian 12. I tried `sudo apt install llvm-15` but it does not seem to contain any permutation of `lldb-vscode`. Did you discover any `lldb-vscode` inside of any official Debian packages? Cheers!

lldb-vscode is part of the [lldb-15 package](https://packages.debian.org/bookworm/amd64/lldb-15/filelist) in stable as well as testing. 
It's located in `/usr/lib/llvm-15/bin/lldb-vscode`

hope that helps!

**Reply by David-Else** [reply]
Thanks, brilliant, cheers!

**Reply by David-Else** [reply]
I added it to the WIKI: https://github.com/helix-editor/helix/wiki/Debugger-Configurations#debian-12-users

**Comment by HealsCodes** [comment]
@David-Else - great job keeping the Wiki article clean and incorporating all the different Linux distributions as they pop-up.

I was wondering if we should grab some external reference like [this one from Microsoft](https://github.com/dotnet/diagnostics/blob/main/documentation/lldb/linux-instructions.md) and copy some of the details the wiki is currently missing (as they have the major ones like Ubuntu/RHEL/Centos/etc)?

Getting all the various Linux distributions that might have a native lldb package under an umbrella feels a bit like fighting windmills these days.

**Comment by janTatesa** [comment]
Any opensuse user here? How to install it for opensuse?

**Reply by HealsCodes** [reply]
same as for most other linux distros - install the lldb package. 
For opensuse that's lldb15 / lldb16 / lldb17 depending which version you want

**Comment by janTatesa** [comment]
> same as for most other linux distros - install the lldb package. For opensuse that's lldb15 / lldb16 / lldb17 depending which version you want

But still I don't have lldb vscode

**Reply by HealsCodes** [reply]
lldb17 installs `/usr/bin/lldb-vscode` and `/usr/bin/lldb-vscode-17` - the **version matters**

*__lldb18 does NOT come with lldb-vscode__*

(checked against lldb17-17.0.6-3.3.x86_64.rpm from OpenSUSE tumbleweed)

**Reply by YatesRocks** [reply]
I was wondering why `zypper in lldb` didn't work as expected on my machine, didn't even think that the newer version stopped supplying it. I should note, on my particular installation `lldb17` installs into `/usr/bin/lldb-vscode-17` _only_, so you need to symlink to `~/.local/bin/lldb-vscode` or edit the Helix config with the full path for it to work

**Comment by giann** [comment]
`brew install llvm` does not give me neither `llvm-dap` or `llvm-vscode`
```shell
$ brew info llvm
==> llvm: stable 18.1.4 (bottled), HEAD [keg-only]
Next-gen compiler infrastructure
https://llvm.org/
Installed
/opt/homebrew/Cellar/llvm/18.1.4 (7,721 files, 1.8GB)
  Poured from bottle using the formulae.brew.sh API on 2024-05-03 at 22:06:22
From: https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/l/llvm.rb

$ ls $(brew --prefix)/opt/llvm/bin/llvm-dap
ls: /opt/homebrew/opt/llvm/bin/llvm-dap: No such file or directory

$ ls $(brew --prefix)/opt/llvm/bin/llvm-vscode
ls: /opt/homebrew/opt/llvm/bin/llvm-vscode: No such file or directory
```

Any idea why?

**Reply by HealsCodes** [reply]
two things come to mind:
- you're looking for *lldb*-vscode not *llvm*-vscode (the package is llvm but you want the debugger not the compiler)
- I noticed that many linux distros also stopped putting lldb-vscode into the llvm/lldb 18 packages so you might have to go back to llvm@17

**Reply by mversic** [reply]
you will find it in `/opt/homebrew/opt/llvm@18/bin/lldb-dap -h`

**Reply by giann** [reply]
Thanks. I was indeed searching for llvm-dap instead of lldb-dap 😩

**Comment by 0x6c6d** [comment]
I had the problem today & fixed it with creating a symlink (suggested here: https://github.com/helix-editor/helix/issues/9964)

lldb-vscode was renamed to lldb-dap, so most likely you will find /usr/bin/lldb-dap in your bin folder after installing lldb (version 18 or later)

This will create a symlink:
`sudo ln -s /usr/bin/lldb-dap /usr/bin/lldb-vscode`

**Reply by HealsCodes** [reply]
> I had the problem today & fixed it with creating a symlink (suggested here: https://github.com/helix-editor/helix/issues/9964)
> 
> 
> 
> lldb-vscode was renamed to lldb-dap, so most likely you will find /usr/bin/lldb-dap in your bin folder after installing lldb (version 18 or later)
> 
> 
> 
> This will create a symlink:
> 
> `sudo ln -s /usr/bin/lldb-dap /usr/bin/lldb-vscode`

Wouldn't the sensible solution be to edit your helix languages.toml and tell it to use lldb-dap instead of relying on symlinks? 

Feels odd to "fix your file system to match what your config file expects"

**Reply by 0x6c6d** [reply]
yes, you're right. I could change my languages.toml based on this commit: https://github.com/helix-editor/helix/commit/b24c465a086b60096d7e95852d4d2db84f5ccbd3

**Comment by bp7968h** [comment]
Hi, as of now `lldb-vscode` have been renamed to `lldb-dap`, so If you are in **linux** especially **fedora** then you can use the below command which installs it:
```bash
sudo dnf install lldb
```
Now this should be auto available in the helix configuration which you can check using the following command:
```bash
hx --health rust
```
>Configured language servers:
  ✓ rust-analyzer: /home/bpandit/.cargo/bin/rust-analyzer
Configured debug adapter: lldb-dap
Binary for debug adapter: /usr/bin/lldb-dap
Configured formatter: rustfmt
Binary for formatter: /home/bpandit/.cargo/bin/rustfmt
Highlight queries: ✓
Textobject queries: ✓
Indent queries: ✓

**Comment by ehrktia** [comment]
late reply,
for any one having issues with `lldb-dap`  in `ubuntu` please install `lldb` via apt 
```bash
sudo apt-get upgrade && sudo apt-get update
sudo apt-get install lldb -y
sudo ln -s /usr/bin/lldb-dap-18 /usr/bin/lldb-dap
```
this will link the lldb-dap to path and add `/usr/bin/lldb-dap` to your `PATH` 
FYI : using `hx` **master** branch
```bash
 hx --health rust
Configured language servers:
  ✓ rust-analyzer: /home/ehrktia/.cargo/bin/rust-analyzer
Configured debug adapter: lldb-dap
Binary for debug adapter: /usr/bin/lldb-dap
Configured formatter: None
Tree-sitter parser: ✓
Highlight queries: ✓
Textobject queries: ✓
Indent queries: ✓
```

**Reply by andreheringer** [reply]
Up, as a add-on for any newbies: the link ehkrita is creating here `sudo ln -s /usr/bin/lldb-dap-18 /usr/bin/lldb-dap` is for lldb-dap version 18 make sure to use the version you have installed. You can check your version with `lldb --version`
